src/Raven.Server/Documents/Subscriptions/SubscriptionProcessor/OrchestratedSubscriptionProcessor.cs

```diff
@@ -43,25 +50,128 @@ public override IEnumerable<(Document Doc, Exception Exception)> GetBatch()
     {
         if (_state.Batches.TryTake(out CurrentBatch, TimeSpan.Zero) == false)
             yield break;
-
+        
         using (CurrentBatch.ReturnContext)
+        using (Server.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
         {
+            var migrationFilter = new SubscriptionBatchMigrationHelper();
+            migrationFilter.Init(Server, _databaseContext.DatabaseName, Connection.SubscriptionId, ShardHelper.GetShardNumber(CurrentBatch.ShardName));
+
             foreach (var batchItem in CurrentBatch.Items)
             {
                 Connection.CancellationTokenSource.Token.ThrowIfCancellationRequested();
 
                 if (batchItem.ExceptionMessage != null)
                     yield return (null, new Exception(batchItem.ExceptionMessage));
+                
+                using (var document = new Document
+                       {
+                           ChangeVector = batchItem.ChangeVector, 
+                           Id = ClusterContext.GetLazyString(batchItem.Id)
+                       })
+                {
+                    var changeVector = context.GetChangeVector(document.ChangeVector);
+                    var shouldSkip = migrationFilter.ShouldSkip(context.Allocator, document.Id, changeVector, out var reason);
+                    if (shouldSkip == false)
+                    {
+                        document.Data = batchItem.RawResult.Clone(ClusterContext);
```

Comment:
What disposes this?

Reply:
the caller do `using (itemDetails.Doc.Data)`